include/deal.II/dofs/dof_tools.h

```diff
@@ -1437,6 +1437,21 @@ namespace DoFTools
                                          const std::set<types::boundary_id> &boundary_ids = std::set<types::boundary_id>());
 
   /**
+   * Extract all degrees of freedom (DoFs) that have support only within cells,
+   * for which @p predicate is <code>true</code>. The result is returned as
```

Comment:
can you add a sentence that says that the predicate will only be called for non-artificial cells (with a link to the glossary entry of "artificial cells") and that the returned set consequently only contains DoFs located on either locally owned or ghost cells?